app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java

```diff
@@ -173,6 +173,15 @@ public List<RuleType> listGlobalRules() {
     @Audited(extractParameters = {"0", KEY_RULE})
     @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
     public void createGlobalRule(Rule data) {
+        String type = data.getType();
+        requireParameter("type", type);
```

Comment:
Why is calling `requireParameter` not sufficient?

Reply:
Because `requireParameter` doesn't check for empty string.

Reply:
That was my point, actually. The `requireParameter` only checks for nullability and, If we want to ensure that the type is not null, we should have it defined at the OpenAPI definition level as not null. Same goes for the config.